### PR TITLE
Add road and structure mapping

### PR DESCRIPTION
**Pull request by borysiasty**
This PR replaces #2719 and solves #2673

### CODE REVIEW

File: safe/gui/tools/wizard_metadata.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+"""
+__author__ = 'qgis@borysjurgiel.pl'
+__revision__ = '$Format:%H$'
+__date__ = '12/04/2016'
+__copyright__ = ('Copyright 2012, Australia Indonesia Facility for '
+                 'Disaster Reduction')
+
+
+road_class_mapping = [{
```
Comment (Gustry) [inline safe/gui/tools/wizard_metadata.py]: Is-it possible to translate these strings ? (at least in the UI, maybe not in the XML)

Comment (ismailsunni) [inline safe/gui/tools/wizard_metadata.py]: Yes, I agree with @Gustry 
We need two thing for each class, `key` and `name`. `key` is the machine-friendly, not translatable. Where `name` is the human friendly one, it should be translatable.

Please take a look at the example below from class mapping in definitions.py

```
    'classes': [
        {
            'key': 'wet',
            'name': tr('wet'),
            'description': tr('Water above ground height.'),
            'numeric_default_min': 1,
            'numeric_default_max': 9999999999,
            'optional': True
        },
        {
            'key': 'dry',
            'name': tr('dry'),
            'description': tr('No water above ground height.'),
            'numeric_default_min': 0,
            'numeric_default_max': (1 - small_number),
            'optional': True
        }
    ]
```
A description also should be added.

I am fine with `value_mapping`. What do you think @timlinux

**Comment by timlinux** [comment]
:+1: for ``value_mapping`` from me

**Comment by borysiasty** [comment]
Ok, I added names and descriptions to the temporary metadata and renamed the " **FINAL** " name value_map once again to value_mapping ;)

File: safe/gui/tools/wizard_metadata.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+from safe.utilities.i18n import tr
+
+road_class_mapping = [{
+    'key': 'Motorway / highway',
+    'name': tr('Motorway / highway'),
+    'description': tr('Motorway / highway'),
+    'string_defaults': ['Motorway or highway']
+}, {
+    'key': 'Motorway link',
+    'nam': 'Motorway link',
```
Comment (Gustry) [inline safe/gui/tools/wizard_metadata.py]: `name`

**Comment by borysiasty** [comment]
Oops, thanks @Gustry !

**Comment by borysiasty** [comment]
(I mean it's fixed now :) )

**Comment by samnawi** [comment]
LGTM.
A feature to mapping the road and structure is work.

I've some input for this :
1. Could we add some word in the balloon tip when the mouse over the classified unique value to explain the meaning of the class? I think it can help the user to decide the best group for the item.
image
2. Maybe it will be better if the column follows the size of the window. So, when the unclassified items were too much, the user can optimize the screen to see it all.
screenshot_1

In The result, I still look this word 
image

Thank you.

**Comment by borysiasty** [comment]
Hi @samnawi,
1. The text for the baloon is taken from the class description. As there was no metadata for the classes yet, I put they to a temporary file until @ismailsunni implements it in a proper place:  https://github.com/borysiasty/inasafe/blob/develop/safe/gui/tools/wizard_metadata.py

I don't know who is the best suited for updating the descriptions. Anyway, the wizard is ready to use the updated description.

2. Unfortunately the height of all steps in the wizard is restricted to 400px. Because of a bug (in QWebView I guess), the MessageViewer embedded in the wizard causes serious problems with resizing, and this was a necessary workaround. Soon the wizard will be refactored (I'll make a PR after this one is merged) and I'll do my best to only apply that restriction to steps containing the MessageViewer. Anyway, the problem is not caused by this PR - that height limit already existed.

**Comment by ismailsunni** [comment]
Hi @borysiasty @samnawi 
Yes, we will need to add some description in the metadata that can be shown in the balloon tip. It's merely adding the description. We can do it later. Or even better, @samnawi can help us what should be the description.

And yes, we need @Gustry to implement the next step as I mentioned in https://github.com/inasafe/inasafe/issues/2673#issuecomment-207191423. My part has been done by @borysiasty  :D 

This PR is LGTM. If no one dis-LGTM it, we can merge it today, and pass the stick to @Gustry 

Thanks @borysiasty and @samnawi

**Comment by Gustry** [comment]
Ok, thanks, I will start look at the next step ;-)

**Comment by timlinux** [comment]
LGTM too thanks @borysiasty !

**Comment by samnawi** [comment]
@borysiasty I have a proposed for structure classification.
Could we refer to this [article ](http://wiki.openstreetmap.org/wiki/Id:Referensi_Atribut_Objek_OSM_Indonesia)to regrouping the structure? It has a good classification and easy to classified, because there is some explanation in there.
**1.1 Pendidikan / Education**
Ini merupakan objek-objek yang memberikan fasilitas pelayanan pendidikan khususnya pendidikan formal seperti sekolah.
An Object that has a service in education sector.
**1.2 Kesehatan / Health**
Objek yang memberikan pelayanan dan fasilitas di bidang kesehatan seperti pengobatan dan perawatan.
An Object that has a service and facility in health sector.
**1.3 Transportasi / Transportation**
Objek yang memberikan pelayanan dan fasilitas transportasi seperti transportasi umum.
An Object that has a service and facility in public transportation.
**1.4 Tempat Ibadah / Place of Worship**
Objek yang digunakan untuk melakukan peribadatan ataupun kegiatan yang berkaitan dengan agama dan keyakinan masyarakat.
An Object that used to pray or related to religion activity.
**1.5 Kantor Pemerintahan / Government**
Bangunan yang digunakan untuk kegiatan pemerintah baik dalam bentuk pelayanan masyarakat maupun kegiatan pemerintahan yang lain.
A building that used to doing government activity in public service or the other government activity.
**1.6 Ekonomi / Economy**
Bangunan yang digunakan untuk kegiatan jual beli terhadap suatu produk/jasa ataupun objek yang didalamnya terdapat aktifitas ekonomi.
A building that used to trade / buy and sell activity or an object that has an economy activity.
**1.7 Rekreasi dan Hiburan / Recreation and Entertainment**
Objek yang menyediakan hiburan dan juga fasilitas rekreasi.
An Object that provide an entertainment or recreation facilities.
**1.8 Olahraga / Sport**
Objek yang menyediakan fasilitas olahraga dan masyarakat bisa menggunakannya untuk kegiatan olahraga.
An Object that has a sport facility and people can use it.
**1.9 Fasilitas Public / Public Facility**
Objek yang menyediakan fasilitas ataupun layanan untuk kegiatan masyarakat. Beberapa contohnya seperti toilet umum, gedung pertemuan, perpustakaan dan lain-lain.
An Object that provide a service or facility to public like toilet, library, convention hall, etc.
**1.10 Akomodasi dan Makanan / Accommodation**
Objek yang menyediakan fasilitas penginapan ataupun makanan dan minuman.
An Object that provide an accommodation / lodging or food services.
**1.11 Bangunan Lain / Other**
Objek lain yang terdapat di Indonesia dan seringkali butuh untuk dipetakan.
An Object that be found in Indonesia, and frequently mapped.

Thank you.

**Comment by Charlotte-Morgan** [comment]
apologies for the delay commenting on this.
The above classification is based on the OSM ID wiki classification. http://wiki.openstreetmap.org/wiki/Id:Referensi_Atribut_Objek_OSM_Indonesia
However; while this may work in Indonesia, it might not fit well for other people.

I would like to propose the following changes:

**Education**
A structure that provides services in the education sector
**Health**
A structure that has a service and facility in the health sector.
**Transport**
A structure that provides services in the transport sector.
**Place of Worship**
A structure that is used for prayer or related to religious activity.
**Government**
A building that used to support government activity or provide a public service.
**Commercial**
A structure that is used for commercial purposes; activities such as trade, buying or selling.
**Industrial**
A structure that is used for manufacturing or industrial purposes
**Recreation**
A structure used for providing entertainment, sporting or recreation facilities.
**Public Facility**
An Object that provides a service or facility to the public like toilet, library, convention hall, etc.
**Residential**
A building that provides a primary place of residence or accommodation.
**Other**
Other built structures that are not specified elsewhere.

**notes:**
all shops, restaurants, hotels etc would be commercial
all factories, manufacturing, power stations, water treatment plants etc are industrial

According to OSM ID these are grouped in Economy but this is not usually the case in other countries.

I am happy to edit https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L971 with some specific examples if people agree.
This document should also reference the source of the information so people know where it comes from.

# cc
@timlinux @easmetz @samnawi

**Comment by harrymahar** [comment]
Hi Charlotte

After I discussed with the team, I think it would be better for InaSAFE to using your proposed classification. I assume that this classification will be used not only in Indonesia but also in other countries. Therefore, it would be better if the classification can be recognized easier by worldwide. Thus, we will edit the Indonesia osm wiki page http://wiki.openstreetmap.org/wiki/Id:Referensi_Atribut_Objek_OSM_Indonesia and our OSM Indonesia Presets based on the classification. 

So for resume, we use your proposed classification and give our OSM Indonesia Wiki page as a reference for the classification.

Notes : All object in each classification only defined as a building (polygon). For Landuse category (Area) we still using same classification including residential and residential area.

Thankyou

## CC
@Charlotte-Morgan @samnawi @easmetz

**Comment by Gustry** [comment]
@Charlotte-Morgan You can edit the file if you want, or I can do it today.

**Comment by Charlotte-Morgan** [comment]
hi @harrymahar  - thank you for considering the suggestion and advice.
@Gustry  - I will edit the file at https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L971

I assume this will be reflected in the wizard.
Is the order of the features in the impact report linked to the order of items in the structure_class_mapping list?

**Comment by Gustry** [comment]
Yes, the order is important. It will be used in the wizard and in the report.

**Comment by Charlotte-Morgan** [comment]
so i can change the order to more logical :)

**Comment by Gustry** [comment]
Of course :)
Don't worry for the `osm_downloader` field, I will update it later. And `string_defaults`, it's a list, the wizard will try to match as much as possible values from the attribute table.

**Pull request merged**